clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
I do not think we should introduce an unnecessary dependency to grpc (i.e., json). Are there particular reasons you want grpc internals receives a json object and decode it instead of receiving a native struct object? If we really need json, I think it should be wrapped somewhere else instead of this place.

Yes, we will add that API to extract the service config.

With the API in this PR, it is very easy to inject a service config the grpc internals. We do not need any test-only API. Notice that this impl does NOT bind service config with name resolver, which is more extensible and flexible.